### DISCUSSION TITLE
Networking and vehicles

### DISCUSSION THREAD

**Discussion by Pindrought** [jrouwe/JoltPhysics#632] [Q&A] [answered]
I am currently going through and trying to figure out how to approach networking using a wheeled vehicle and getting a barebones demo working.

With the basic character test, it was very simple to get working and I only am transmitting the character position/velocity/movement vector.

With the vehicle though, I am wondering what all needs to be taken into consideration.

I am assuming I will be sending the following inputs...
**-forward/right/brake/handbrake input values**

I am a bit confused on how I should be sending over the current values for things like speed/current vehicle position.
Should I just be using
**GetLinearVelocity,GetPosition and GetRotation and using bodyInterface.SetPositionRotationAndVelocity for the body associated with the vehicle?**

I am mainly wondering if there are variables other than this that I should be taking into consideration.

Thanks!

**Comment by jrouwe** [comment, marked as answer]
I think the answer depends on what your general strategy is for replication. If you rely on a deterministic simulation and you follow all the rules for making the simulation deterministic between server and client then you only needs to replicate the inputs (forward, right, brake, handbrake), but if your simulation is not deterministic then you are going to have to send positions/orientations/velocities using the functions you mention.

**Reply by Pindrought** [reply]
Thank you it ended up working great!
